server.js

```diff
@@ -1,23 +1,110 @@
 const express = require("express");
 const cors = require("cors");
+const bodyParser = require("body-parser");
 
 const app = express();
 
-app.use(cors());
+app.use(cors({
+        origin: 'http://localhost:3000',
+    }
+));
+
+
+app.use(bodyParser.urlencoded({ extended: true }));
 
 const welcomeMessage = {
   id: 0,
   from: "Bart",
   text: "Welcome to CYF chat system!",
+  // timeSent:"18/05/1985"
 };
 
 //This array is our "data store".
 //We will start with one message in the array.
 //Note: messages will be lost when Glitch restarts our server.
-const messages = [welcomeMessage];
+let messages = [welcomeMessage]
+
+
+// level3 [5] read only text whose text contains a given substring
+// teniolao-cyf-chat-server.glitGlitchch.me/messages/search?text=express
+app.get("/messages/search", (req, res)=> {
+  const { term } = req.query
+  console.log(term)
+  
+  const filterMessages = messages.filter(message => message.text.toLowerCase().includes(term.toLowerCase()))
+  console.log(filterMessages)
+  res.send(filterMessages);
+});
+
+//[2] read all messages
+app.get("/messages", (req, res) => {
+  res.json(messages);
+});
+//  Read only the most recent 10 messages:
+
+app.get("/messages/latest",(req,res)=>{
+  res.send(messages.slice(-10))
+})
+
+//[3] read one message specified by id
+app.get("/messages/:id", function (req, res) {
+  const id = req.params.id;
+  messages = messages.filter(
+    (message) => message.id === Number(id)
+  );
+    res.status(200).json(messages);
+});
+
+app.get("/messages/search",(req,res)=>{
+  let search=req.query.term
+  let searchResult=messages.filter(
+    mes=>mes.text.toLowerCase().includes(search.toLowerCase())
+  )
+  res.send(searchResult)
+})
+
+// [1] create a new message
 
-app.get("/", function (request, response) {
-  response.sendFile(__dirname + "/index.html");
+app.post("/messages", (req, res) => {
+  const { from, text } = req.body;
+  // console.log(req.body)
+  const ourMessageObject = {
```

Comment:
this method of selecting an id can lead to duplicate id values.

Imaging you had a set of 20 messages, where the last message is id:20. You delete message 19. When you next send a message, it will also be given id:20